update parameters & return value & exceptions of Notification constructor (#31706)
mdn/content: files/en-us/web/api/notification/notification/index.md

        active until the user clicks or dismisses it, rather than closing automatically.
        The default value is `false`.
    - `actions` {{optional_inline}}

      - : An array of actions to display in the notification which the default is an empty list. Each element in the array is an object with the following members:

        - `action`
          - : A string identifying a user action to be displayed on the notification.
        - `title`
          - : A string containing action text to be shown to the user.
        - `icon` {{optional_inline}}
          - : A string containing the URL of an icon to display with the action.

        Appropriate responses are built using `event.action` within the
        {{domxref("ServiceWorkerGlobalScope.notificationclick_event", "notificationclick")}} event.

    - `silent` {{optional_inline}}
      - : A boolean value specifying whether the
        notification is silent (no sounds or vibrations issued), regardless of the device
        settings. The default is `null`. If `true`, then `vibrate` must not be present.

### Return value

An instance of the {{domxref("Notification")}} object.

### Exceptions

- {{jsxref("TypeError")}}

>   - : Thrown if:
>     - The constructor is called within the {{domxref("ServiceWorkerGlobalScope")}}.
>     - The `silent` option is `true` and the `vibrate` option is specified.
>     - The `renotify` option is `true` but the `tag` option is empty.
> - `DataCloneError` {{domxref("DOMException")}}
>   - : Thrown if serialize the `data` option failed for some reason.

Author: ```suggestion
  - : Thrown if:
    - The constructor is called within the {{domxref("ServiceWorkerGlobalScope")}}.
    - The `actions` option is specified and is not empty.
    - The `silent` option is `true` and the `vibrate` option is specified.
    - The `renotify` option is `true` but the `tag` option is empty.
- `DataCloneError` {{domxref("DOMException")}}
  - : Thrown if serialize the `data` option failed for some reason.
```

      - : a string containing the URL of an image to
        be displayed in the notification.
    - `data` {{optional_inline}}
      - : Arbitrary data that you want associated with the
        notification. This can be of any data type.
        The default is `null`.
    - `vibrate` {{optional_inline}}
      - : A [vibration pattern](/en-US/docs/Web/API/Vibration_API#vibration_patterns) for the device's vibration hardware to emit with the notification. If specified, `silent` must not be `true`.
    - `timestamp` {{optional_inline}}
      - : A number representing the time at which a notification is created or applicable (past, present, or future).
    - `renotify` {{optional_inline}}
      - : A boolean value specifying whether the user should be notified after a new notification replaces an old one. The default is `false`, which means they won't be notified. If `true`, then `tag` also must be set.
    - `requireInteraction` {{optional_inline}}
      - : Indicates that a notification should remain
        active until the user clicks or dismisses it, rather than closing automatically.
        The default value is `false`.
    - `actions` {{optional_inline}}

>       - : An array of actions to display in the notification which the default is an empty list. Each element in the array is an object with the following members:

Reviewer: ```suggestion
      - : An array of actions to display in the notification, for which the default is an empty array. Each element in the array can be an object with the following members:
```

          - : A string identifying a user action to be displayed on the notification.
        - `title`
          - : A string containing action text to be shown to the user.
        - `icon` {{optional_inline}}
          - : A string containing the URL of an icon to display with the action.

        Appropriate responses are built using `event.action` within the
        {{domxref("ServiceWorkerGlobalScope.notificationclick_event", "notificationclick")}} event.

    - `silent` {{optional_inline}}
      - : A boolean value specifying whether the
        notification is silent (no sounds or vibrations issued), regardless of the device
        settings. The default is `null`. If `true`, then `vibrate` must not be present.

### Return value

An instance of the {{domxref("Notification")}} object.

### Exceptions

- {{jsxref("TypeError")}}
  - : Thrown if:
    - The constructor is called within the {{domxref("ServiceWorkerGlobalScope")}}.
    - The `actions` option is specified and is not empty.
    - The `silent` option is `true` and the `vibrate` option is specified.
    - The `renotify` option is `true` but the `tag` option is empty.
- `DataCloneError` {{domxref("DOMException")}}

>   - : Thrown if serialize the `data` option failed for some reason.

Reviewer: ```suggestion
  - : Thrown if serializing the `data` option failed for some reason.
```

>   - : Thrown if:
>     - The constructor is called within the {{domxref("ServiceWorkerGlobalScope")}}.
>     - The constructor is not called within the {{domxref("ServiceWorkerGlobalScope")}} and the `actions` option is not empty.
>     - The `silent` option is `true` and the `vibrate` option is specified.
>     - The `renotify` option is `true` but the `tag` option is empty.
> - `DataCloneError` {{domxref("DOMException")}}
>   - : Thrown if serialize the `data` option failed for some reason.

Author: ```suggestion
  - : Thrown if:
    - The constructor is called within the {{domxref("ServiceWorkerGlobalScope")}}.
    - The `silent` option is `true` and the `vibrate` option is specified.
    - The `renotify` option is `true` but the `tag` option is empty.
- `DataCloneError` {{domxref("DOMException")}}
  - : Thrown if serialize the `data` option failed for some reason.
```
